### PR TITLE
Optimise images

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR

- Refactors  autonomy and agent images
  - Include install and build scripts in the base image
  - Remove unwanted layers
  - Remove unwanted data files

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes
- [ ] Public-facing documentation has been updated with the changes affected by this PR. Even if the provided contents are not in their final form, all significant information must be included.
- [ ] Any backwards-incompatible/breaking change has been clearly documented in the upgrading document.

### CODE REVIEW

File: autonomy/data/Dockerfiles/agent/Dockerfile [hunk 1]
```diff
@@ -6,15 +6,12 @@ FROM ${AUTONOMY_IMAGE_NAME}:${AUTONOMY_IMAGE_VERSION}
 ARG AEA_AGENT
 ARG AUTHOR
 
-COPY start.sh /home/ubuntu
-COPY install.sh /home/ubuntu
```
Comment (angrybayblade) [inline autonomy/data/Dockerfiles/agent/Dockerfile:10]: Will be included in the base image

File: deployments/Dockerfiles/autonomy/Dockerfile [hunk 2]
```diff
@@ -2,25 +2,8 @@ ARG AEA_VERSION=latest
 
 FROM valory/open-aea-user:${AEA_VERSION}
 
+RUN apt install git net-tools sudo -y
 
-WORKDIR /home/ubuntu
-ENV PATH=$PATH:/home/ubuntu/.local/bin
+COPY scripts /root/scripts
```
Comment (angrybayblade) [inline deployments/Dockerfiles/autonomy/Dockerfile:7]: Install and runner scripts

File: deployments/Dockerfiles/autonomy/Dockerfile [hunk 3]
```diff
@@ -2,25 +2,8 @@ ARG AEA_VERSION=latest
...
 
-RUN sudo apt-get update --fix-missing
-RUN sudo apt-get autoremove
-RUN sudo apt-get autoclean
-RUN sudo apt-get install python3.10 python3.10-dev git cmake autoconf automake build-essential libffi-dev libtool pkg-config -y
-RUN cd /usr/bin && sudo rm python3 && sudo ln -s python3.10 python3 && sudo ln -s python3.10 python
```
Comment (angrybayblade) [inline deployments/Dockerfiles/autonomy/Dockerfile:13]: No need to install the python interpreters and dev libraries, they are included in the base image

File: deployments/Dockerfiles/autonomy/Dockerfile [hunk 4]
```diff
@@ -2,25 +2,8 @@ ARG AEA_VERSION=latest
...
-
-# used in Docker-compose to wait until Hardhat node is up
-RUN sudo apt-get install net-tools -y
-RUN python -m pip install --upgrade pip
-
-COPY openssl.cnf /etc/ssl
```
Comment (angrybayblade) [inline deployments/Dockerfiles/autonomy/Dockerfile:19]: Included in the base image

File: deployments/Dockerfiles/autonomy/requirements.txt [hunk 5]
```diff
@@ -1,4 +0,0 @@
-open-aea[all]>=1.40.0,<2.0.0
-open-aea-cli-ipfs>=1.40.0,<2.0.0
-protobuf>=4.21.6,<5.0.0
-requests==2.27.1
```
Comment (angrybayblade) [inline deployments/Dockerfiles/autonomy/requirements.txt:4]: Already included in the base image

**Review by solarw — APPROVED** [review]
LGTM

**Review by dvilelaf — APPROVED** [review]

**Pull request merged**